[V21] Incorrect version prefix for vaadin-core after package.json update

TL;DR: the `package.json` contains version with `.` instead of `-`:

```
"@vaadin/vaadin-core": "21.0.0.alpha1"
```

### Description of the bug / feature

I’m updating docs to use `21.0.0.alpha1` and faced this problem:

```
npm ERR! code ETARGET
npm ERR! notarget No matching version found for @vaadin/vaadin-core@21.0.0.alpha1.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
npm ERR! notarget
npm ERR! notarget It was specified as a dependency of 'docs'
npm ERR! notarget
```

See the diff of changes to package.json after the update:

<details>
<summary>Full diff</summary>

```diff
diff --git a/package.json b/package.json
index 97e95e2c..cb52af55 100644
--- a/package.json
+++ b/package.json
@@ -5,47 +5,47 @@
     "dependencies": {
       "@vaadin/router": "1.7.4",
       "@polymer/polymer": "3.2.0",
-      "@vaadin/vaadin-crud": "20.0.0-beta1",
-      "@vaadin/vaadin-icons": "20.0.0-beta1",
-      "@vaadin/vaadin-grid": "20.0.0-beta1",
-      "@vaadin/vaadin-split-layout": "20.0.0-beta1",
-      "@vaadin/vaadin-combo-box": "20.0.0-beta1",
-      "@vaadin/vaadin-cookie-consent": "20.0.0-beta1",
-      "@vaadin/vaadin-upload": "20.0.0-beta1",
-      "@vaadin/vaadin-dialog": "20.0.0-beta1",
-      "@vaadin/vaadin-select": "20.0.0-beta1",
-      "@vaadin/vaadin-app-layout": "20.0.0-beta1",
-      "@vaadin/vaadin-item": "20.0.0-beta1",
-      "@vaadin/vaadin-board": "20.0.0-beta1",
-      "@vaadin/vaadin-notification": "20.0.0-beta1",
-      "@vaadin/vaadin-charts": "20.0.0-beta1",
-      "@vaadin/vaadin-grid-pro": "20.0.0-beta1",
-      "@vaadin/vaadin-progress-bar": "20.0.0-beta1",
-      "@vaadin/vaadin-date-time-picker": "20.0.0-beta1",
-      "@vaadin/vaadin-ordered-layout": "20.0.0-beta1",
-      "@vaadin/vaadin-login": "20.0.0-beta1",
-      "@vaadin/vaadin-button": "20.0.0-beta1",
-      "@vaadin/vaadin-date-picker": "20.0.0-beta1",
-      "@vaadin/vaadin-text-field": "20.0.0-beta1",
-      "@vaadin/vaadin-menu-bar": "20.0.0-beta1",
-      "@vaadin/vaadin-custom-field": "20.0.0-beta1",
-      "@vaadin/vaadin-form-layout": "20.0.0-beta1",
-      "@vaadin/vaadin-confirm-dialog": "20.0.0-beta1",
-      "@vaadin/vaadin-accordion": "20.0.0-beta1",
+      "@vaadin/vaadin-crud": "21.0.0-alpha2",
+      "@vaadin/vaadin-icons": "21.0.0-alpha2",
+      "@vaadin/vaadin-grid": "21.0.0-alpha2",
+      "@vaadin/vaadin-split-layout": "21.0.0-alpha2",
+      "@vaadin/vaadin-combo-box": "21.0.0-alpha2",
+      "@vaadin/vaadin-cookie-consent": "21.0.0-alpha2",
+      "@vaadin/vaadin-upload": "21.0.0-alpha2",
+      "@vaadin/vaadin-dialog": "21.0.0-alpha2",
+      "@vaadin/vaadin-select": "21.0.0-alpha2",
+      "@vaadin/vaadin-app-layout": "21.0.0-alpha2",
+      "@vaadin/vaadin-item": "21.0.0-alpha2",
+      "@vaadin/vaadin-board": "21.0.0-alpha2",
+      "@vaadin/vaadin-notification": "21.0.0-alpha2",
+      "@vaadin/vaadin-charts": "21.0.0-alpha2",
+      "@vaadin/vaadin-grid-pro": "21.0.0-alpha2",
+      "@vaadin/vaadin-progress-bar": "21.0.0-alpha2",
+      "@vaadin/vaadin-date-time-picker": "21.0.0-alpha2",
+      "@vaadin/vaadin-ordered-layout": "21.0.0-alpha2",
+      "@vaadin/vaadin-login": "21.0.0-alpha2",
+      "@vaadin/vaadin-button": "21.0.0-alpha2",
+      "@vaadin/vaadin-date-picker": "21.0.0-alpha2",
+      "@vaadin/vaadin-text-field": "21.0.0-alpha2",
+      "@vaadin/vaadin-menu-bar": "21.0.0-alpha2",
+      "@vaadin/vaadin-custom-field": "21.0.0-alpha2",
+      "@vaadin/vaadin-form-layout": "21.0.0-alpha2",
+      "@vaadin/vaadin-confirm-dialog": "21.0.0-alpha2",
+      "@vaadin/vaadin-accordion": "21.0.0-alpha2",
       "@polymer/iron-list": "3.1.0",
-      "@vaadin/vaadin-list-box": "20.0.0-beta1",
-      "@vaadin/vaadin-checkbox": "20.0.0-beta1",
-      "@vaadin/vaadin-details": "20.0.0-beta1",
+      "@vaadin/vaadin-list-box": "21.0.0-alpha2",
+      "@vaadin/vaadin-checkbox": "21.0.0-alpha2",
+      "@vaadin/vaadin-details": "21.0.0-alpha2",
       "@polymer/iron-icon": "3.0.1",
-      "@vaadin/vaadin-time-picker": "20.0.0-beta1",
-      "@vaadin/vaadin-context-menu": "20.0.0-beta1",
-      "@vaadin/vaadin-radio-button": "20.0.0-beta1",
-      "@vaadin/vaadin-tabs": "20.0.0-beta1",
-      "@vaadin/vaadin-lumo-styles": "20.0.0-beta1",
-      "@vaadin/vaadin-material-styles": "20.0.0-beta1",
-      "@vaadin/vaadin-rich-text-editor": "20.0.0-beta1",
-      "@vaadin/vaadin-avatar": "20.0.0-beta1",
-      "@vaadin/vaadin-messages": "20.0.0-beta1",
+      "@vaadin/vaadin-time-picker": "21.0.0-alpha2",
+      "@vaadin/vaadin-context-menu": "21.0.0-alpha2",
+      "@vaadin/vaadin-radio-button": "21.0.0-alpha2",
+      "@vaadin/vaadin-tabs": "21.0.0-alpha2",
+      "@vaadin/vaadin-lumo-styles": "21.0.0-alpha2",
+      "@vaadin/vaadin-material-styles": "21.0.0-alpha2",
+      "@vaadin/vaadin-rich-text-editor": "21.0.0-alpha2",
+      "@vaadin/vaadin-avatar": "21.0.0-alpha2",
+      "@vaadin/vaadin-messages": "21.0.0-alpha2",
       "lit": "2.0.0-rc.1",
       "@polymer/iron-a11y-announcer": "3.0.2",
       "@polymer/iron-a11y-keys-behavior": "3.0.1",
@@ -58,13 +58,18 @@
       "@polymer/iron-resizable-behavior": "3.0.1",
       "@polymer/iron-scroll-target-behavior": "3.0.1",
       "@webcomponents/shadycss": "1.9.6",
-      "@vaadin/vaadin-control-state-mixin": "20.0.0-beta1",
+      "@vaadin/vaadin-control-state-mixin": "21.0.0-alpha2",
       "@vaadin/vaadin-development-mode-detector": "2.0.4",
-      "@vaadin/vaadin-element-mixin": "20.0.0-beta1",
-      "@vaadin/vaadin-list-mixin": "20.0.0-beta1",
-      "@vaadin/vaadin-overlay": "20.0.0-beta1",
-      "@vaadin/vaadin-themable-mixin": "20.0.0-beta1",
-      "@vaadin/vaadin-usage-statistics": "2.1.0"
+      "@vaadin/vaadin-element-mixin": "21.0.0-alpha2",
+      "@vaadin/vaadin-list-mixin": "21.0.0-alpha2",
+      "@vaadin/vaadin-overlay": "21.0.0-alpha2",
+      "@vaadin/vaadin-themable-mixin": "21.0.0-alpha2",
+      "@vaadin/vaadin-usage-statistics": "2.1.0",
+      "@vaadin/vaadin-core-shrinkwrap": "21.0.0-alpha1",
+      "@vaadin/vaadin-shrinkwrap": "21.0.0-alpha1",
+      "@vaadin/vaadin-template-renderer": "21.0.0-alpha2",
+      "@vaadin/vaadin-virtual-list": "21.0.0-alpha2",
+      "@vaadin/vaadin-core": "21.0.0.alpha1"
     },
     "devDependencies": {
       "compression-webpack-plugin": "4.0.1",
@@ -92,7 +97,7 @@
       "ts-loader": "8.0.12",
       "fork-ts-checker-webpack-plugin": "6.2.1"
     },
-    "hash": "0c81b06dbbedba1f96e99c389b5e7a37cecd8eaff1b64e7b8f9c7aeb071c32ee"
+    "hash": "ce15191bddb8096d9b7f0b0955d12ed12b8e6415ad58a6171e721a02ddae61d6"
   },
   "dependencies": {
     "@polymer/iron-a11y-announcer": "3.0.2",
@@ -111,57 +116,62 @@
     "@vaadin/flow-frontend": "./target/flow-frontend",
     "@vaadin/form": "./target/flow-frontend/form",
     "@vaadin/router": "1.7.4",
-    "@vaadin/vaadin-accordion": "20.0.0-beta1",
-    "@vaadin/vaadin-app-layout": "20.0.0-beta1",
-    "@vaadin/vaadin-avatar": "20.0.0-beta1",
-    "@vaadin/vaadin-board": "20.0.0-beta1",
-    "@vaadin/vaadin-button": "20.0.0-beta1",
-    "@vaadin/vaadin-charts": "20.0.0-beta1",
-    "@vaadin/vaadin-checkbox": "20.0.0-beta1",
-    "@vaadin/vaadin-combo-box": "20.0.0-beta1",
-    "@vaadin/vaadin-confirm-dialog": "20.0.0-beta1",
-    "@vaadin/vaadin-context-menu": "20.0.0-beta1",
-    "@vaadin/vaadin-control-state-mixin": "20.0.0-beta1",
-    "@vaadin/vaadin-cookie-consent": "20.0.0-beta1",
-    "@vaadin/vaadin-crud": "20.0.0-beta1",
-    "@vaadin/vaadin-custom-field": "20.0.0-beta1",
-    "@vaadin/vaadin-date-picker": "20.0.0-beta1",
-    "@vaadin/vaadin-date-time-picker": "20.0.0-beta1",
-    "@vaadin/vaadin-details": "20.0.0-beta1",
+    "@vaadin/vaadin-accordion": "21.0.0-alpha2",
+    "@vaadin/vaadin-app-layout": "21.0.0-alpha2",
+    "@vaadin/vaadin-avatar": "21.0.0-alpha2",
+    "@vaadin/vaadin-board": "21.0.0-alpha2",
+    "@vaadin/vaadin-button": "21.0.0-alpha2",
+    "@vaadin/vaadin-charts": "21.0.0-alpha2",
+    "@vaadin/vaadin-checkbox": "21.0.0-alpha2",
+    "@vaadin/vaadin-combo-box": "21.0.0-alpha2",
+    "@vaadin/vaadin-confirm-dialog": "21.0.0-alpha2",
+    "@vaadin/vaadin-context-menu": "21.0.0-alpha2",
+    "@vaadin/vaadin-control-state-mixin": "21.0.0-alpha2",
+    "@vaadin/vaadin-cookie-consent": "21.0.0-alpha2",
+    "@vaadin/vaadin-crud": "21.0.0-alpha2",
+    "@vaadin/vaadin-custom-field": "21.0.0-alpha2",
+    "@vaadin/vaadin-date-picker": "21.0.0-alpha2",
+    "@vaadin/vaadin-date-time-picker": "21.0.0-alpha2",
+    "@vaadin/vaadin-details": "21.0.0-alpha2",
     "@vaadin/vaadin-development-mode-detector": "2.0.4",
-    "@vaadin/vaadin-dialog": "20.0.0-beta1",
-    "@vaadin/vaadin-element-mixin": "20.0.0-beta1",
-    "@vaadin/vaadin-form-layout": "20.0.0-beta1",
-    "@vaadin/vaadin-grid": "20.0.0-beta1",
-    "@vaadin/vaadin-grid-pro": "20.0.0-beta1",
-    "@vaadin/vaadin-icons": "20.0.0-beta1",
-    "@vaadin/vaadin-item": "20.0.0-beta1",
-    "@vaadin/vaadin-list-box": "20.0.0-beta1",
-    "@vaadin/vaadin-list-mixin": "20.0.0-beta1",
-    "@vaadin/vaadin-login": "20.0.0-beta1",
-    "@vaadin/vaadin-lumo-styles": "20.0.0-beta1",
-    "@vaadin/vaadin-material-styles": "20.0.0-beta1",
-    "@vaadin/vaadin-menu-bar": "20.0.0-beta1",
-    "@vaadin/vaadin-messages": "20.0.0-beta1",
-    "@vaadin/vaadin-notification": "20.0.0-beta1",
-    "@vaadin/vaadin-ordered-layout": "20.0.0-beta1",
-    "@vaadin/vaadin-overlay": "20.0.0-beta1",
-    "@vaadin/vaadin-progress-bar": "20.0.0-beta1",
-    "@vaadin/vaadin-radio-button": "20.0.0-beta1",
-    "@vaadin/vaadin-rich-text-editor": "20.0.0-beta1",
-    "@vaadin/vaadin-select": "20.0.0-beta1",
-    "@vaadin/vaadin-split-layout": "20.0.0-beta1",
-    "@vaadin/vaadin-tabs": "20.0.0-beta1",
-    "@vaadin/vaadin-text-field": "20.0.0-beta1",
-    "@vaadin/vaadin-themable-mixin": "20.0.0-beta1",
-    "@vaadin/vaadin-time-picker": "20.0.0-beta1",
-    "@vaadin/vaadin-upload": "20.0.0-beta1",
+    "@vaadin/vaadin-dialog": "21.0.0-alpha2",
+    "@vaadin/vaadin-element-mixin": "21.0.0-alpha2",
+    "@vaadin/vaadin-form-layout": "21.0.0-alpha2",
+    "@vaadin/vaadin-grid": "21.0.0-alpha2",
+    "@vaadin/vaadin-grid-pro": "21.0.0-alpha2",
+    "@vaadin/vaadin-icons": "21.0.0-alpha2",
+    "@vaadin/vaadin-item": "21.0.0-alpha2",
+    "@vaadin/vaadin-list-box": "21.0.0-alpha2",
+    "@vaadin/vaadin-list-mixin": "21.0.0-alpha2",
+    "@vaadin/vaadin-login": "21.0.0-alpha2",
+    "@vaadin/vaadin-lumo-styles": "21.0.0-alpha2",
+    "@vaadin/vaadin-material-styles": "21.0.0-alpha2",
+    "@vaadin/vaadin-menu-bar": "21.0.0-alpha2",
+    "@vaadin/vaadin-messages": "21.0.0-alpha2",
+    "@vaadin/vaadin-notification": "21.0.0-alpha2",
+    "@vaadin/vaadin-ordered-layout": "21.0.0-alpha2",
+    "@vaadin/vaadin-overlay": "21.0.0-alpha2",
+    "@vaadin/vaadin-progress-bar": "21.0.0-alpha2",
+    "@vaadin/vaadin-radio-button": "21.0.0-alpha2",
+    "@vaadin/vaadin-rich-text-editor": "21.0.0-alpha2",
+    "@vaadin/vaadin-select": "21.0.0-alpha2",
+    "@vaadin/vaadin-split-layout": "21.0.0-alpha2",
+    "@vaadin/vaadin-tabs": "21.0.0-alpha2",
+    "@vaadin/vaadin-text-field": "21.0.0-alpha2",
+    "@vaadin/vaadin-themable-mixin": "21.0.0-alpha2",
+    "@vaadin/vaadin-time-picker": "21.0.0-alpha2",
+    "@vaadin/vaadin-upload": "21.0.0-alpha2",
     "@vaadin/vaadin-usage-statistics": "2.1.0",
     "@webcomponents/shadycss": "1.9.6",
     "date-fns": "^2.16.1",
     "lit": "2.0.0-rc.1",
     "promise-file-reader": "^1.0.3",
-    "vanilla-colorful": "^0.5.3"
+    "vanilla-colorful": "^0.5.3",
+    "@vaadin/vaadin-core-shrinkwrap": "21.0.0-alpha1",
+    "@vaadin/vaadin-shrinkwrap": "21.0.0-alpha1",
+    "@vaadin/vaadin-template-renderer": "21.0.0-alpha2",
+    "@vaadin/vaadin-virtual-list": "21.0.0-alpha2",
+    "@vaadin/vaadin-core": "21.0.0.alpha1"
   },
   "devDependencies": {
     "@types/validator": "13.1.0",
```

</details>

### Minimal reproducible example

1. `git clone https://github.com/vaadin/docs` 
2. `git checkout lit2-update`
3. Update version in pom.xml to be `21.0.0.alpha1`
4. Run `mvn`

### Expected behavior

The version should look like this (using `-`):

```
"@vaadin/vaadin-core": "21.0.0-alpha1"
```

### Actual behavior

The version looks like this (using `.`):

```
"@vaadin/vaadin-core": "21.0.0.alpha1"
```

### Versions:
    - Vaadin / Flow version: 21.0.0.alpha1

--- a/flow-server/src/main/java/com/vaadin/flow/server/frontend/VersionsJsonConverter.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/frontend/VersionsJsonConverter.java
@@ -15,6 +15,8 @@
  */
 package com.vaadin.flow.server.frontend;
 
+import java.util.Objects;
+
 import elemental.json.Json;
 import elemental.json.JsonObject;
 import elemental.json.JsonValue;
@@ -32,6 +34,7 @@ import elemental.json.JsonValue;
  */
 class VersionsJsonConverter {
 
+    static final String VAADIN_CORE_NPM_PACKAGE = "@vaadin/vaadin-core";
     private static final String JS_VERSION = "jsVersion";
     private static final String NPM_NAME = "npmName";
     private static final String NPM_VERSION = "npmVersion";
@@ -71,6 +74,10 @@ class VersionsJsonConverter {
     private void addDependency(JsonObject obj) {
         assert obj.hasKey(NPM_NAME);
         String npmName = obj.getString(NPM_NAME);
+        // #11025
+        if (Objects.equals(npmName, VAADIN_CORE_NPM_PACKAGE)) {
+            return;
+        }
         if (obj.hasKey(NPM_VERSION)) {
             convertedObject.put(npmName, obj.getString(NPM_VERSION));
         } else if (obj.hasKey(JS_VERSION)) {

--- a/flow-server/src/test/java/com/vaadin/flow/server/frontend/TaskUpdatePackagesNpmTest.java
+++ b/flow-server/src/test/java/com/vaadin/flow/server/frontend/TaskUpdatePackagesNpmTest.java
@@ -37,8 +37,6 @@ import org.junit.Test;
 import org.junit.rules.ExpectedException;
 import org.junit.rules.TemporaryFolder;
 import org.mockito.Mockito;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 import com.vaadin.flow.server.Constants;
 import com.vaadin.flow.server.frontend.scanner.ClassFinder;
@@ -51,6 +49,7 @@ import elemental.json.JsonValue;
 import static com.vaadin.flow.server.Constants.PACKAGE_JSON;
 import static com.vaadin.flow.server.frontend.NodeUpdater.DEPENDENCIES;
 import static com.vaadin.flow.server.frontend.NodeUpdater.DEV_DEPENDENCIES;
+import static com.vaadin.flow.server.frontend.VersionsJsonConverter.VAADIN_CORE_NPM_PACKAGE;
 import static com.vaadin.flow.server.frontend.NodeUpdater.VAADIN_DEP_KEY;
 
 @NotThreadSafe
@@ -423,6 +422,46 @@ public class TaskUpdatePackagesNpmTest {
 
         verifyVersions(PLATFORM_DIALOG_VERSION, PLATFORM_ELEMENT_MIXIN_VERSION,
                 PLATFORM_OVERLAY_VERSION);
+    }
+
+    // #11025
+    @Test
+    public void npmIsInUse_versionsJsonHasVaadinCoreVersionPinned_vaadinCoreVersionIgnored()
+            throws IOException {
+        final String expectedElementMixinVersion = "21.0.0-alpha2";
+        String versionJsonString = //@formatter:off
+                "{ \"core\": {"
+                        + "\"vaadin-element-mixin\": {\n"
+                        + "    \"jsVersion\": \""+expectedElementMixinVersion+"\",\n"
+                        + "    \"npmName\": \""+VAADIN_ELEMENT_MIXIN+"\"\n"
+                        + "},\n"
+                        + "\"vaadin-core\": {\n"
+                        + "    \"jsVersion\": \"21.0.0.alpha1\",\n" // broken for npm
+                        + "    \"npmName\": \""+VAADIN_CORE_NPM_PACKAGE+"\"\n"
+                        + "},\n"
+                        +"}}},\n";//@formatter:on
+        FileUtils.write(versionJsonFile, versionJsonString,
+                StandardCharsets.UTF_8);
+
+        final TaskUpdatePackages task = createTask(
+                createApplicationDependencies());
+        task.execute();
+        Assert.assertTrue("Updates not picked", task.modified);
+
+        verifyVersions(PLATFORM_DIALOG_VERSION, expectedElementMixinVersion,
+                null);
+        final JsonObject packageJson = getOrCreatePackageJson();
+        JsonObject dependencies = packageJson.getObject(DEPENDENCIES);
+
+        Assert.assertFalse(
+                VAADIN_CORE_NPM_PACKAGE
+                        + " version should not be written to package.json",
+                dependencies.hasKey(VAADIN_CORE_NPM_PACKAGE));
+        final JsonObject vaadinDependencies = packageJson
+                .getObject(VAADIN_DEP_KEY).getObject(DEPENDENCIES);
+        Assert.assertFalse(VAADIN_CORE_NPM_PACKAGE
+                + " version should not be written to vaadin dependencies in package.json",
+                vaadinDependencies.hasKey(VAADIN_CORE_NPM_PACKAGE));
     }
 
     private void createBasicVaadinVersionsJson() {

--- a/flow-server/src/test/java/com/vaadin/flow/server/frontend/VersionsJsonConverterTest.java
+++ b/flow-server/src/test/java/com/vaadin/flow/server/frontend/VersionsJsonConverterTest.java
@@ -25,6 +25,8 @@ import org.junit.rules.TemporaryFolder;
 import elemental.json.Json;
 import elemental.json.JsonObject;
 
+import static com.vaadin.flow.server.frontend.VersionsJsonConverter.VAADIN_CORE_NPM_PACKAGE;
+
 public class VersionsJsonConverterTest {
 
     @Rule
@@ -52,7 +54,11 @@ public class VersionsJsonConverterTest {
                     "            \"javaVersion\": \"3.0.0.beta1\",\n" +
                     "            \"jsVersion\": \"2.0.19\"\n" +
                     "        },"
-                    +"\"platform\": \"foo\""
+                  + "\"vaadin-core\": {\n" +
+                    "    \"jsVersion\": \"21.0.0.alpha1\",\n" + // broken for npm
+                    "    \"npmName\": \""+VAADIN_CORE_NPM_PACKAGE+"\"\n" +
+                    "},\n"
+                  +"\"platform\": \"foo\""
                 + "}";
         // @formatter:on
 
@@ -65,6 +71,7 @@ public class VersionsJsonConverterTest {
 
         Assert.assertFalse(convertedJson.hasKey("flow"));
         Assert.assertFalse(convertedJson.hasKey("core"));
+        Assert.assertFalse(convertedJson.hasKey(VAADIN_CORE_NPM_PACKAGE));
         Assert.assertFalse(convertedJson.hasKey("platform"));
 
         Assert.assertEquals("1.1.2",

